build.gradle

```diff
@@ -0,0 +1,331 @@
+	implementation files("$jarDir/amazon/V2/json-utils-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/metrics-spi-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/profiles-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/protocol-core-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/regions-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/s3-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/sdk-core-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/third-party-jackson-core-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/url-connection-client-2.17.214.jar")
+	implementation files("$jarDir/amazon/V2/utils-2.17.214.jar")
+	implementation files("$jarDir/reactivestreams/reactive-streams-1.0.3.jar")
+
+	// TEST
+	testImplementation files("$jarDir/apache_commons/commons-collections-3.2.1.jar")
+	testImplementation files("$jarDir/apache_commons/commons-lang-2.4.jar")
+	testImplementation files("$jarDir/apache_commons/velocity-1.7.jar")
+	testImplementation files("$jarDir/arangodb/arangodb-java-driver-6.7.2.jar")
+	testImplementation files("$jarDir/arangodb/velocypack-2.4.0.jar")
+	testImplementation files("$jarDir/bytebuddy/byte-buddy-1.9.10.jar")
+	testImplementation files("$jarDir/bytebuddy/byte-buddy-agent-1.9.10.jar")
+	testImplementation files("$jarDir/equalsverifier/equalsverifier-3.1.10.jar")
+	testImplementation files("$jarDir/jsemver/java-semver-0.9.0.jar")
+	testImplementation files("$jarDir/kbase/auth2/kbase-auth2test-0.2.4.jar")
+	testImplementation files("$jarDir/mockito/mockito-core-3.0.0.jar")
+	testImplementation files("$jarDir/objenesis/objenesis-2.6.jar")
+}
+
+// download the KBase jars if they aren't already present
+task downloadJars {
+	doLast {
+		def dir = file(downloadDir)
+		if (!dir.exists()) {
+			dir.mkdirs()
+		}
+		def url = "https://github.com/kbase/jars/raw/master/lib/jars/kbase"
+		def jars = [
+			"auth/kbase-auth-0.4.4.jar",
+			"auth2/kbase-auth2test-0.2.4.jar",
+			"common/$kbaseCommonJar",
+			"handle/AbstractHandleClient-1.0.0.jar",
+			"kidl/kbase-kidl-parser-1409261812-7863aef.jar",
+			"sample/SampleServiceClient-0.1.1.jar",
+			"shock/shock-client-0.1.0.jar"
+		]
+		jars.each { jar ->
+			def jarName = jar.split("/")[-1]
+			def file = file("$downloadDir/$jarName")
+			if (!file.exists()) {
+				ant.get(src: "$url/$jarName", dest: "$downloadDir/$jarName")
+			}
+		}
+	}
+}
+
+// Custom java project layout
+sourceSets {
+	main {
+		java {
+			srcDir src
+			exclude "**/test/**"
+		}
+	}
+	test {
+		java {
+			srcDir src
+		}
+		resources {
+			srcDir src
+			include "**/*.properties"
+			include "**/*.cfg"
+			include "**/*.spec"
+			include "**/*.instance.*"
+			include "**/instance.*"
+			include "**/*.instance"
+			include "**/*.html"
+			include "**/*.css"
+			include "**/*.gif"
+			include "**/*.js"
+			include "**/*.png"
+			include "**/*.txt"
+			include "**/*.weirdsuffix"
+		}
+	}
+}
+
+tasks.withType(JavaCompile) {
```

Comment:
## Compilation

Run `gradle compileJava` to compile the source code excluding tests for the workspace, or `gradle compileTestJava` to compile everything. The results can be found in the directory `build/classes/`.